src/app/reporting/ReportScheduler.h

```diff
@@ -167,14 +188,39 @@ class ReportScheduler : public ReadHandler::Observer
 
     virtual void ReportTimerCallback() = 0;
 
+    /// @brief Method that triggers a report emission on each ReadHandler that are above their min interval
+    ///        It checks if all read handlers are above their min interval and if so forces them in a dirty state to trigger a
+    ///        report emission
```

Comment:
```suggestion
    /// @brief Method that triggers a report emission on each ReadHandler that is not blocked on its min interval.
    ///        Each read handler that is not blocked is immediately marked dirty so that it will report as soon as possible.
```